source-git/README.md

```diff
@@ -0,0 +1,89 @@
+# Source-Git
+
+## How to do the rebase via source-git?
+
+Generally, we need to provide the following tasks:
+
+- [ ] The new change needs to be converted from upstream to source-git.
+- [ ] The changed source-git can be converted to dist-git.
+  - Via PR or a new Bugzilla.
+  - (This text does not cover this.)
+
+What changes can occur?
+
+- Change in upstream means always change in the source-archive in distgit.
+- Change in source-git can mean:
+  - Change in distribution files => can be converted directly.
+  - Change in the files => new patch needs to be added.
+
+### Solutions for upstream -> source-git from GIT perspective
+
+#### 1. New branch for each release
+
+- We have one branch (`upstream` in the following image) mapping the upstream releases.
+- For each new upstream release, we will create a new source-git branch.
+
+<img src="./img/sync-upstream-release.svg" alt="sync-upstream-release" height="200"/>
+
+#### 2. Force-push
+
+- For each new upstream release, we will regenerate source-git and force-push.
+
+<img src="./img/force-push-upstream-release.svg" alt="force-push-upstream-release" height="200"/>
+
+#### 3. Revert & Regenerate
+
+- We can use same solution we discussed [updates from distgit](../dist-git-to-src-git/updates.md).
+
+<img src="./img/revert-and-regenerate-upstream-release.svg" alt="revert-and-regenerate-upstream-release" height="200"/>
+
```

Comment:
how about doing the new update in a dedicated branch and then git-merging it?

it also seems like things will be different when the source-git repo has an upstream history and when it doesn't

Reply:
> how about doing the new update in a dedicated branch and then git-merging it?

Will this work?

* Can we mix source-git commits with the upstream ones?
* We need to have full git-history for the generation. (Will not be preserved by merge.)

Reply:
for 9 it will definitely work

for 8 it seems recreate is the way to go - I wonder if we can move some of the code from d2s to packit to do as many manual steps as possible

would you have time to update this research wrt difference b/w 8 and 9 source-git repos?

Reply:
Sorry, I missed your last comment and not sure if I get it correctly. The following scenario is not supported by packit, is it? (Or a different question, do we want to support it?)

```
EXAMPLE SOURCE-GIT HISTORY:

* downstream-patch commit
* upstream commit
* upstream commit
* downstream-patch commit
* downstream packaging files
* upstream commit
:
```

Reply:
depends what you mean by supported, upstream_ref is the king here - whatever is behind, packit doesn't care, whatever is ahead, it will be processed

so you're right that git-merge is not the way to go, we either need to force-push (recreate) or create branches per rebase

Reply:
@TomasTomecek I'm a little bit confused now: shouldn't this be only about 9, and that is only about source-git repos which connect to the upstream history? What am I missing?

Reply:
> supported

Packit can work with such repo => srpm/rpm can be created & bugzilla/PR to distgit can be generated